deps/graph-parser/src/logseq/graph_parser/util/block_ref.cljs

```diff
@@ -0,0 +1,41 @@
+(ns logseq.graph-parser.util.block-ref
+  "General purpose vars and util fns for block-refs"
+  (:require [clojure.string :as string]))
+
+(def left-parens "Opening characters for block-ref" "((")
+(def right-parens "Closing characters for block-ref" "))")
+(def left-and-right-parens "Opening and closing characters for block-ref"
+  (str left-parens right-parens))
+(def block-ref-re #"\(\(([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})\)\)")
+
+(defn get-all-block-ref-ids
+  [content]
+  (map second (re-seq block-ref-re content)))
+
+(defn get-block-ref-id
```

Comment:
It's interesting to see that we have two ways of extracting and checking for block-refs. String based ones are called `get-string-block-ref-id` and `string-block-ref?` and regex-based ones are `get-block-ref-id` and `block-ref?`. They serve different purposes as string fns are less strict since they are used with user inputs